File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Yunyinghuodong.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Yunyinghuodong implements ITableOperate {
 public final static String TAB_FILE_DATA = "/yunyinghuodong.txt";
 public static final int ID_DECRIPTION = 0 ;
public static final int ID_TRIGGER_TYPE = 1 ;
public static final int ID_TRIGGER_VALUE = 2 ;
public static final int ID_END_TYPE = 3 ;
public static final int ID_END_VALUE = 4 ;
public static final int ID_REWARD_TYPE = 5 ;
public static final int ID_REWARD_VALUE = 6 ;
public static final int ID_REWARD_NUM = 7 ;
public static final int MAX_RECORD = 8 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_Decription;
 public String GetDecription() { return m_Decription;}

private int m_EndType;
 public int GetEndType() { return m_EndType;}

private String m_EndValue;
 public String GetEndValue() { return m_EndValue;}

private String m_RewardNum;
 public String GetRewardNum() { return m_RewardNum;}

private String m_RewardType;
 public String GetRewardType() { return m_RewardType;}

private String m_RewardValue;
 public String GetRewardValue() { return m_RewardValue;}

private int m_TriggerType;
 public int GetTriggerType() { return m_TriggerType;}

private int m_TriggerValue;
 public int GetTriggerValue() { return m_TriggerValue;}

public boolean LoadTable(HashMap<Integer, Table_Yunyinghuodong> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Yunyinghuodong> _hash = (HashMap<Integer, Table_Yunyinghuodong> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Yunyinghuodong _values = new Table_Yunyinghuodong();
 _values.m_Decription =  (String)valuesList.get(ID_DECRIPTION);
_values.m_EndType =  Integer.parseInt((String)valuesList.get(ID_END_TYPE));
_values.m_EndValue =  (String)valuesList.get(ID_END_VALUE);
_values.m_RewardNum =  (String)valuesList.get(ID_REWARD_NUM);
_values.m_RewardType =  (String)valuesList.get(ID_REWARD_TYPE);
_values.m_RewardValue =  (String)valuesList.get(ID_REWARD_VALUE);
_values.m_TriggerType =  Integer.parseInt((String)valuesList.get(ID_TRIGGER_TYPE));
_values.m_TriggerValue =  Integer.parseInt((String)valuesList.get(ID_TRIGGER_VALUE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Yunyinghuodong.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Yunyinghuodong : ITableOperate
{ private const string TAB_FILE_DATA = "yunyinghuodong.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_DECRIPTION,
ID_TRIGGER_TYPE,
ID_TRIGGER_VALUE,
ID_END_TYPE,
ID_END_VALUE,
ID_REWARD_TYPE,
ID_REWARD_VALUE,
ID_REWARD_NUM,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_Decription;
 public string Decription { get{ return m_Decription;}}

private int m_EndType;
 public int EndType { get{ return m_EndType;}}

private string m_EndValue;
 public string EndValue { get{ return m_EndValue;}}

private string m_RewardNum;
 public string RewardNum { get{ return m_RewardNum;}}

private string m_RewardType;
 public string RewardType { get{ return m_RewardType;}}

private string m_RewardValue;
 public string RewardValue { get{ return m_RewardValue;}}

private int m_TriggerType;
 public int TriggerType { get{ return m_TriggerType;}}

private int m_TriggerValue;
 public int TriggerValue { get{ return m_TriggerValue;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Yunyinghuodong _values = new Tab_Yunyinghuodong();
 _values.m_Decription =  valuesList[(int)_ID.ID_DECRIPTION] as string;
_values.m_EndType =  Convert.ToInt32(valuesList[(int)_ID.ID_END_TYPE] as string);
_values.m_EndValue =  valuesList[(int)_ID.ID_END_VALUE] as string;
_values.m_RewardNum =  valuesList[(int)_ID.ID_REWARD_NUM] as string;
_values.m_RewardType =  valuesList[(int)_ID.ID_REWARD_TYPE] as string;
_values.m_RewardValue =  valuesList[(int)_ID.ID_REWARD_VALUE] as string;
_values.m_TriggerType =  Convert.ToInt32(valuesList[(int)_ID.ID_TRIGGER_TYPE] as string);
_values.m_TriggerValue =  Convert.ToInt32(valuesList[(int)_ID.ID_TRIGGER_VALUE] as string);

 _hash.Add(nKey,_values); }


}
}

